src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -195,4 +195,65 @@ object OrderedRVDPartitioner {
       i
     }
   }
+
+  // takes range bounds from n partitioners and splits them such that each
+  // resulting partition only comes from one partition per original partitioner.
+  def mergePartitioners(p1: OrderedRVDPartitioner, p2: OrderedRVDPartitioner): OrderedRVDPartitioner = {
+    require(p1.pkType == p2.pkType)
```

Comment:
I think that this assertion should really be comparing keys, not partition keys. See #4040 for a bug caused by a similar assertion. (It's confusing right now, but I'm working to clean things up.)